pkg/podstate/podstate.go

```diff
@@ -38,6 +38,16 @@ func (s *State) GetPodFromState(podName string, podNamespace string) powerv1.Gua
 	return powerv1.GuaranteedPod{}
 }
 
+func (s *State) GetPodFromStateUID(podUID string) powerv1.GuaranteedPod {
```

Comment:
Kubernetes already enforces that a pod name/namespace combination is unique so why is the existing GetPodFromState function not sufficient?

Reply:
IMO this part of the change is unnecessary and adds complexity by introducing a second way to retrieve the pod.

Reply:
I fixed these modifications per your suggestion.